assets/differentialabundance_report.Rmd

```diff
@@ -382,19 +389,43 @@ differential_results <- lapply(differential_files, function(diff_file){
         qval_column = params$differential_qval_column
     )
 
-    # If fold changes are not logged already, log them (we assume they're logged
-    # later on)
-
-    if (! params$differential_foldchanges_logged){
+    # If fold changes are not logged already, log them
+    if (!params$differential_foldchanges_logged) {
         diff[[params$differential_fc_column]] <- log2(diff[[params$differential_fc_column]])
     }
 
     # Annotate differential tables if possible
-    if (! is.null(params$features)){
-        diff <- merge(features, diff, by.x = params$features_id_col, by.y = params$differential_feature_id_column)
+    if (!is.null(params$features)) {
+
+        # Merge tables
+        diff_features <- merge(features, diff, by.x = params$features_id_col, by.y = params$differential_feature_id_column)
+
+        # Get number of rows before and after merging
+        length_diff <- as.numeric(nrow(diff))
+        length_features <- as.numeric(nrow(diff_features))
+
+        # Compare numbers and report
+        if (length_diff != length_features) {
```

Comment:
@pinin4fjords I agree that this should be addressed really upstream (as mentioned on the PR description), but the rmd code should contain the error handling as well, in case anyone runs it alone. What do you think?

Reply:
Yeah, fair enough.